src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -631,11 +658,83 @@ internal static void HighlightPort(Step stepInfo, StepUIAutomation uiAutomationD
                 if (buttonRectangle != null)
                     mainGrid.Children.Remove(buttonRectangle);
             }
-           
+
         }
 
         /// <summary>
-        /// This method will be called when is necessary to detect a node creation command from the workspace
+        /// This method will enable or disable the Next Popup Button and also subcribe to the PropertyChanged event so we will know if the Nodes are connected or not
+        /// </summary>
+        /// <param name="stepInfo">Information about the Step</param>
+        /// <param name="uiAutomationData">Information about UI Automation that is being executed</param>
+        /// <param name="enableFunction">Variable used to know if we are executing the automation or undoing changes</param>
+        /// <param name="currentFlow">Current Guide Flow</param>
+        internal static void ManagePortConnectionEvents(Step stepInfo, StepUIAutomation uiAutomationData, bool enableFunction, GuideFlow currentFlow)
+        {
+            CurrentExecutingStep = stepInfo;
+
+            var portHighlighted = uiAutomationData.Parameters[1] as string;
+            if (portHighlighted == null) return;
+
+            //Get the button name to be enabled/disabled coming from the json file
+            var buttonName = CurrentExecutingStep.UIAutomation[1].Parameters[0] as string;
+            if (buttonName == null) return;
+
+            //Get the Popup and search for the NextButton in the VisualTree
+            var popupWindow = CurrentExecutingStep.StepUIPopup as PopupWindow;
+            var buttonFound = GuideUtilities.FindChild(popupWindow.mainPopupGrid, buttonName) as Button;
+            if (buttonFound == null) return;
+
+            var stepMainWindow = CurrentExecutingStep.MainWindow as Window;
+
+            var byOriginNotConnected = GetNotConnectedNode(stepMainWindow, "CoordinateSystem.ByOrigin");
+            if (byOriginNotConnected == null) return;
+
+            var idString = byOriginNotConnected.ViewModel.Id.ToString().Replace("-","");
```

Comment:
same here

Reply:
@QilongTang fixed to only use ToString() method in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12860/commits/ab9dac8beced65ebb949df6a3d59919273f40a45